azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/model/Runtime.java

```diff
@@ -34,6 +37,91 @@
     public static final Runtime LINUX_JAVA11_TOMCAT85 = new Runtime(OperatingSystem.LINUX, WebContainer.TOMCAT_85, JavaVersion.JAVA_11);
     public static final Runtime DOCKER = new Runtime(OperatingSystem.DOCKER, null, null);
 
+    public static Runtime getRuntime(OperatingSystem operatingSystem, WebContainer webContainer, JavaVersion javaVersion) {
+        return values().stream()
+                .filter(runtime -> runtime.operatingSystem == operatingSystem)
+                .filter(runtime -> runtime.webContainer == webContainer)
+                .filter(runtime -> runtime.javaVersion == javaVersion)
+                .findFirst().orElse(new Runtime(operatingSystem, webContainer, javaVersion));
+    }
+
+    public static Runtime getRuntimeFromWebApp(WebAppBase webAppBase) {
+        if (StringUtils.startsWithIgnoreCase(webAppBase.linuxFxVersion(), "docker")) {
+            return DOCKER;
+        }
+        return webAppBase.operatingSystem() == com.azure.resourcemanager.appservice.models.OperatingSystem.WINDOWS ?
+                getRuntimeFromWindowsWebApp(webAppBase) : getRuntimeFromLinuxWebApp(webAppBase);
+    }
+
+    public static Runtime getRuntimeFromLinuxWebApp(WebAppBase webAppBase) {
```

Comment:
I think this should be webapp's responsibility, i.e. `webapp.getRuntime()`
webapp should only expose a method named `getRuntime`, the webapp returns proper runtime base on its own case, it may be a linux or docker or java app. while the interface/exposed/public method should hide such details.

Reply:
Yes, these methods is not designed to expose to ui, but used by managers as they may be re-used by multi-manager like webapp and deploymentslot, will move to singleton utils